src/PlaneWaveBasis.jl

```diff
@@ -543,6 +565,29 @@ function weighted_ksum(basis::PlaneWaveBasis, array)
     mpi_sum(res, basis.comm_kpts)
 end
 
+"""
+Utilities to get information about the irreducible k-point mesh (in case of duplication)
+Useful for I/O, where k-point information should not be duplicated
+"""
+function irreducible_kcoords(basis::PlaneWaveBasis)
+    # Assume that duplicated k-points are appended at the end of the kcoords array
+    basis.kcoords_global[1:basis.n_irreducible_kpoints]
+end
+
+function irreducible_kweights(basis::PlaneWaveBasis{T}) where {T}
+    irr_kweights = basis.kweights_global[1:basis.n_irreducible_kpoints]
+
+    # Assume that duplicated k-points are appended at the end of the kcoords/kweights array
+    for i_dupl in basis.n_irreducible_kpoints+1:length(basis.kweights_global)
+        for i_irr in 1:basis.n_irreducible_kpoints
+            if maximum(abs.(basis.kcoords_global[i_dupl]-basis.kcoords_global[i_irr])) < eps(T)
```

Comment:
A few comments:
- `maximum(abs, ... )` is better (no allocation).
- Hmm. This mechanics may be a little brittle. I'd be much more defensive here, e.g. make sure each `i_irr` only matches a single kpoint, e.g. by using `only` in combination with `findall` or so. Similarly I'd put an assertion that the weights add up to the total electron count or so (see how it's done in other places, the correct number depends on the computational setup).

Also too long a line :smile:.